File: haskell/fizzbuzz.hs
fizzify :: Int -> String
fizzify n | n `mod` 15 == 0  = "FizzBuzz"
          | n `mod` 5  == 0  = "Buzz"
          | n `mod` 3  == 0  = "Fizz"
          | otherwise        = show n

fizzbuzz :: Int -> [String]
fizzbuzz n = map fizzify [1..n]

main = do
    print(fizzbuzz 10)

File: rust/fizzbuzz.rs
fn fizzify(n) {
    if (n % 15 == 0)
        return "FizzBuzz";
    if (n % 5 == 0)
        return "Buzz";
    if (n % 3 == 0)
        return "Fizz";
    return n.to_string();
}

fn fizzbuzz(n) {
    let mut result = vec![];
    for (let i = 1; i <= n; i++) {
        result.push(fizzify(i));
    }
    return result;
}

// driver code
fn main() {
    println!("{:?}", fizzbuzz(10));
}